lib/change_stream.js

```diff
@@ -132,20 +157,26 @@ var createChangeStreamCursor = function(self) {
   return changeStreamCursor;
 };
 
-var buildChangeStreamAggregationCommand = function(
-  serverConfig,
-  namespace,
-  pipeline,
-  resumeToken,
-  options
-) {
+var buildChangeStreamAggregationCommand = function(self) {
+  const serverConfig = self.serverConfig;
+  const namespace = self.namespace;
+  const pipeline = self.pipeline;
+  const resumeToken = self.resumeToken;
+  const options = self.options;
+  const cursorNamespace = self.cursorNamespace;
+
+  const isMaster = serverConfig.lastIsMaster() || {};
```

Comment:
you're already pulling this off on line 76, why not pass it in?

Reply:
My thoughts are as follows:

1. The spec requires that we get a Timestamp ASAP, and stick with it. So I am getting the Timestamp in the constructor, and keeping it. This is especially important for reconnection, as I need to use the original timestamp.
2. The state of the topology can change over time, so we should use an `ismaster` that is as recent as possible for anything else, including the wire protocol versions.

I'm not sure if that is overthinking it.

Reply:
okay, I think we can throw this in the "once the SDAM refactor comes in we won't worry about stuff like this" bucket 😜